### ISSUE TITLE
Support returning non-zero exit codes on errors only

### ISSUE THREAD

**Issue by RickMoynihan** [clj-kondo/clj-kondo#1259]
### Problem 

It'd be nice to run `git rebase --interactive origin/master --exec "clj-kondo --lint ./src"` and have it fail on configured linters that issue an `:error`.

Unfortunately because of the way the shell interacts with git and the various pieces of the environment it's not easy to find a one-liner that can massage the return code and correctly abort the rebase at the failing commit.

### Suggested Solution 

Hence it would be nice to have a `--fail-on error` flag, which would only return a non-zero status code if there are any lint failures at `:error` level.  The proposed (explicitly stated) default for this would be `--fail-on warning`.

Then the above could be written like this:

`git rebase --interactive origin/master --exec "clj-kondo --lint ./src --fail-on error"`

Without having to create an intermediate script like this:

```shell
#!/bin/bash

clj-kondo --lint ./src

if [ $? -le 2 ]; then
        exit 0
else
        exit 1
fi
```

**Comment by borkdude** [comment]
Rubocop:

bozhidar:cider: We have something like this, let me look it up.
It's fail-level - https://docs.rubocop.org/rubocop/usage/basic_usage.html#command-line-flags
4:16 PM
I think that's pretty common across linters.

Let's go for this then: `--fail-level error`.
